python/xorbits/_mars/utils.py

```diff
@@ -1676,7 +1676,12 @@ def _get_func_token_values(func):
             tokens.extend([func.args, func.keywords])
             func = func.func
         if hasattr(func, "__code__"):
-            tokens.extend(_get_func_token_values(func))
+            if len(func.__code__.co_code) > 0:
+                tokens.extend(_get_func_token_values(func))
+            else:
+                # for new python version, cython function has __code__,
+                # but it's an empty string.
+                tokens.extend([func.__module__, func.__name__])
```

Comment:
I think `func.__qualname__` is better than `func.__name__` for generating token.
e.g.
```python
def foo():
    def bar():
        return
    return bar

b = foo()
b.__name__
'bar'
b.__qualname__
'foo.<locals>.bar'
```

Reply:
To keep the previous behavior, use module and name for generating token.